http/src/main/java/io/micronaut/http/filter/ContentDispositionFilter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.filter;
+
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.annotation.Filter;
+import jakarta.inject.Singleton;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+
+/**
+ * A filter that adds the `Content-Disposition` header to HTTP responses for specific endpoints.
+ */
+@Singleton
+@Filter("/**")
+public class ContentDispositionFilter implements HttpServerFilter {
```

Comment:
Use `@ServerFilter` API

Reply:
In fact I'm not sure we can have a general filter like this. It will affect performance of all controllers, even those without the annotation. Potentially this change needs to go into ResponseLifecycle instead (after #11342 )